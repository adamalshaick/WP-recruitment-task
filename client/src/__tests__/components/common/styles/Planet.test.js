import React from "react";
import renderer from "react-test-renderer";
import "jest-styled-components";
import {
  PlanetCard,
  PlanetWrapper,
  CardHeader
} from "../../../../components/common/styles/Planet";

describe("render component", () => {
  it("renders card", () => {
    const component = renderer.create(<PlanetCard />).toJSON();
    expect(component).toMatchSnapshot();
  });
  it("renders wrapper", () => {
    const component = renderer.create(<PlanetWrapper />).toJSON();
    expect(component).toMatchSnapshot();
  });
  it("renders header", () => {
    const component = renderer.create(<CardHeader />).toJSON();
    expect(component).toMatchSnapshot();
  });
});

describe("props test when active", () => {
  it("shows planet card when active", () => {
    const component = renderer.create(<PlanetCard active={true} />).toJSON();
    expect(component).toHaveStyleRule("opacity", "1");
    expect(component).toHaveStyleRule("visibility", "visible");
  });
});

describe("props test when disabled", () => {
  it("shows planet card when disabled", () => {
    const component = renderer.create(<PlanetCard active={false} />).toJSON();
    expect(component).toHaveStyleRule("opacity", "0");
    expect(component).toHaveStyleRule("visibility", "hidden");
  });
});
