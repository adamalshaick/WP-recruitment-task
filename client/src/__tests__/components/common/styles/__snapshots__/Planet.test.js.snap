// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render component renders card 1`] = `
.c0 {
  position: absolute;
  left: 50%;
  top: 60%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  background-color: white;
  opacity: 0;
  visibility: hidden;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
  padding: 2rem;
  box-shadow: 0px 0px 5px 0px rgba(0,0,0,0.35);
  border-radius: 5px;
  z-index: 2;
  width: 80%;
}

<tr
  className="c0"
/>
`;

exports[`render component renders header 1`] = `
.c0 {
  margin: 1rem;
  font-size: 20px;
  font-weight: 500;
}

<header
  className="c0"
/>
`;

exports[`render component renders wrapper 1`] = `
.c0 {
  width: 100%;
  height: 100%;
  background-color: white;
  opacity: 0;
  visibility: hidden;
  position: fixed;
  top: 0;
  left: 0;
  -webkit-filter: brightness(90%);
  filter: brightness(90%);
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
}

<tr
  className="c0"
/>
`;
